### ISSUE TITLE
Render Separator should use React.memo

### ISSUE THREAD

**Issue by Hirbod** [gorhom/react-native-sticky-item#8]
https://github.com/gorhom/react-native-sticky-item/blob/master/src/StickyItemFlatList.tsx#L194

Since it should be a „Pure Component“.
I would also think that the second parameter for „arePropsEqual“ could always return true in this case.

After I memoized this I could finally get rid of this VirtualizedList is large and slow to update message.

@gorhom

**Comment by Hirbod** [comment]
Maybe like so

```tsx
function separatorPropsEqual() {
  return true;
}
const renderSeparator = React.memo(
  () => <View style={{ width: SEPARATOR_SIZE }} />,
  separatorPropsEqual,
);
```

**Comment by gorhom** [comment]
hi @Hirbod , i'll work on it this weekend

**Comment by Hirbod** [comment]
Maybe the separator should also be overwritable? And I am still feeling like decelerationRate should also be overwritable.

**Comment by gorhom** [comment]
@Hirbod have you tried to optimising the flatlist with `windowSize` & `maxToRenderPerBatch` ?

**Comment by Hirbod** [comment]
@gorhom yes, but after diggin through a lot of posts I found out that also separator should be a Pure/Memoized Component

**Comment by gorhom** [comment]
okay, i can see the importance of this, i'll push it with the next release 👍

**Comment by Hirbod** [comment]
Thanks. You could actually copy over my snippet (and format to your needs). Its working fine. Still thinking if passing own separator is important or not but I don't think that it should be, since the size is adjustable

**Comment by gorhom** [comment]
hmm i'll investigate it more, however for having a memoized separator, you could submit a pr :)

**Comment by Hirbod** [comment]
Sure.

**Comment by Hirbod** [comment]
PR sent #9

**Comment by gorhom** [comment]
thanks , i'll review it now :)

**Comment by Hirbod** [comment]
I think the equal function check should be moved outside of the functional component

**Comment by gorhom** [comment]
released with [v1.3.0](https://github.com/gorhom/react-native-sticky-item/releases/tag/v1.3.0)
